events/api/src/main/java/org/projectnessie/events/api/ContentStoredEvent.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.events.api;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.immutables.value.Value;
+import org.projectnessie.model.Content;
+
+/**
+ * Event that is emitted when a content is stored. This event corresponds to a PUT operation in a
+ * commit, merge or transplant. This event is emitted after the content has been stored.
+ */
+@Schema(
+    type = SchemaType.OBJECT,
+    description =
+        "An event that is emitted when a content is stored. "
+            + "This event corresponds to a PUT operation in a commit, merge or transplant. "
+            + "This event is emitted after the content has been stored.")
+@Value.Immutable
+@JsonTypeName("CONTENT_STORED")
+@JsonSerialize
+@JsonDeserialize
+public interface ContentStoredEvent extends ContentEvent {
+  @Override
+  @Value.Default
+  default EventType getType() {
+    return EventType.CONTENT_STORED;
+  }
+
+  /** The content that was stored. */
+  Content getContent();
```

Comment:
Probably worth to ensure that content (de)serialization happens in the same way as elsewhere, so allowing to (de)serialize a `GenericContent`